go/pkg/tool/tool.go

```diff
@@ -326,6 +326,20 @@ func (c *Client) DownloadDirectory(ctx context.Context, rootDigest, path string)
 	return err
 }
 
+// UploadDirectory uploads a directory from the specified path as a Merkle-tree to the remote cache.
+func (c *Client) UploadDirectory(ctx context.Context, path string) error {
+	log.Infof("Computing Merkle tree rooted at %s", path)
+	root, blobs, stats, err := c.GrpcClient.ComputeMerkleTree(ctx, path, "", "", &command.InputSpec{Inputs: []string{"."}}, filemetadata.NewNoopCache())
```

Comment:
I just realized, the current implementation serializes all hashing, then all uploading. Both should be happening concurrently for effective performance.

Reply:
That's not supported in the `client` package. It's supported in the `cas` package, but that one does not have feature parity with the implementation in the `client` package which is why it's not integrated yet. We are working on converging the two such that streaming (concurrently digest and upload) and batching (digest all, upload all) workflows are possible. No ETAs for the time being though.

Reply:
We're observing a good 90s to hash our content so if this can be prioritized it would be very appreciated. Thanks